c_plus_plus/src/model/utils/time.cpp

```diff
@@ -0,0 +1,11 @@
+#include "time.h"
+
+namespace Model
+{
+qint64 Time::milliseconds(const QString &hour, const QString &minute, const QString &second, const QString &millisecond)
+{
+    return hour.toLongLong() * 60 * 60 * 1000 + minute.toLongLong() * 60 * 1000 + second.toLongLong() * 1000 +
+           millisecond.toLongLong();
```

Comment:
We should check if the strings are empty, and maybe return -1 if so.

Reply:
If the strings are empty, or even if the strings are not numbers, `toLongLong()` will return 0. I think that's what we want here.

Reply:
👍